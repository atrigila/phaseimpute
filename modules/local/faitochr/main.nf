process FAITOCHR {
    tag "$meta.id"
    label 'process_single'

    input:
    tuple val(meta), path(fai), val(addchr)

    output:
    tuple val(meta), path("*.txt"), emit: annot_chr
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    # Take the fai file and add the chr prefix to the chromosome names
    if [ "${addchr}" = true ]; then
        col1=""
        col2="chr"
    else
        col1="chr"
        col2=""
    fi
    awk -F'\t' '{print \$1}'  ${fai} | \
        sed 's/chr//g' | \
        awk -v col1=\${col1} -v col2=\${col2} 'BEGIN {OFS=" "} {print col1\$1, col2\$1}' > ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awk: \$(awk --version | grep -o 'GNU Awk [0-9.]*' | cut -d ' ' -f 3)
    END_VERSIONS
    """

    stub:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awk: \$(awk --version | grep -o 'GNU Awk [0-9.]*' | cut -d ' ' -f 3)
    END_VERSIONS
    """
}
