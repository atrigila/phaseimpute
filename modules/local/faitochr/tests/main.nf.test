nextflow_process {

    name "Test Process FAITOCHR"
    script "../main.nf"
    process "FAITOCHR"

    tag "modules"
    tag "modules_local"
    tag "faitochr"

    test("fai add chr") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/genome.fasta.fai", checkIfExists: true),
                    true
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("fai remove chr") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/genome.fasta.fai", checkIfExists: true),
                    false
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
