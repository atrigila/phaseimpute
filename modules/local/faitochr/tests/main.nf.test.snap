{
    "fai add chr": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,dc360653d0d1848e6cc01661dbff389c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0d85e18b9c36aa2db49ad51930d9a5e6"
                ],
                "annot_chr": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,dc360653d0d1848e6cc01661dbff389c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0d85e18b9c36aa2db49ad51930d9a5e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-08T17:33:55.801913"
    },
    "fai remove chr": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,c8aa996df2a03384617fff85d911f401"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0d85e18b9c36aa2db49ad51930d9a5e6"
                ],
                "annot_chr": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,c8aa996df2a03384617fff85d911f401"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0d85e18b9c36aa2db49ad51930d9a5e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-08T17:34:01.901705"
    }
}
